app/src/main/java/com/github/houseorganizer/houseorganizer/MainScreenActivity.java

```diff
@@ -88,40 +82,59 @@ protected void onCreate(Bundle savedInstanceState) {
         setUpTaskList();
     }
 
-    private void getCurrentHousehold(){
-        String householdId = getIntent().getStringExtra(HOUSEHOLD);
-        TextView text = findViewById(R.id.last_button_activated);
+    private void loadData() {
+        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
+        String householdId = sharedPreferences.getString(CURRENT_HOUSEHOLD, "");
 
-        if (householdId != null) {
-            currentHouse = db.collection("households").document(householdId);
-            text.setText("currentHouse: " + currentHouse.getId());
-        } else {
-            db.collection("households")
-                    .whereArrayContains("residents", mUser.getEmail()).get()
-                    .addOnCompleteListener(task -> {
-                        if (task.isSuccessful()) {
-                            ArrayList<String> households = new ArrayList<String>();
-                            for (QueryDocumentSnapshot document : task.getResult())
+        loadHousehold(householdId);
+    }
+
+    private void loadHousehold(String householdId) {
+        db.collection("households").whereArrayContains("residents", mUser.getEmail()).get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        ArrayList<String> households = new ArrayList<String>();
+                        for (QueryDocumentSnapshot document : task.getResult()) {
+                            if (householdId.equals(document.getId())) {
                                 households.add(document.getId());
```

Comment:
Isn't there a way to instantly get the household document from the id ? 
like db.collection("households").document(householdId).get() instead of the whereArrayContains